### PR TITLE
Added setHSLAColor in addition of shiftHSLAColor.

### LINKED ISSUE #841: shiftHSLAColor, but it is set instead of shift
image

It would be nice to have a function to set color, instead of shifting.

For example, I want to change the opacity (hsla alpha).

### PR DESCRIPTION
**Pull request by DieselTripleX**
## Description

closes #841

Added setHSLAColor function and unit-test for it, as requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/services/color-config/color-functions.ts [hunk 1]
```diff
@@ -45,6 +45,28 @@ export const shiftHSLAColor = (color: ColorInput, offset: { h?: number; s?: numb
   return result.HSLA
 }
 
+export const setHSLAColor = (color: ColorInput, newColor: { h?: number; s?: number; l?: number; a?: number }) => {
+  const result = new ColorTranslator(color)
+
+  if (newColor.h !== undefined) {
+    result.setH(newColor.h)
```
Comment (m0ksem) [inline packages/ui/src/services/color-config/color-functions.ts:52]: Would be nice to write this code in one line and decrease function length.

File: packages/ui/src/services/color-config/tests/color-functions.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { setHSLAColor } from '../color-functions'
+
+describe('ColorFunctions', () => {
+  describe('setHSLAColor', () => {
+    it('changeHEXhue', () => {
```
Comment (m0ksem) [inline packages/ui/src/services/color-config/tests/color-functions.spec.ts:5]: Rename test names to human language. Like "Like change hue in hex color".

**Review by m0ksem — CHANGES_REQUESTED** [review]

File: packages/ui/src/services/color-config/tests/color-functions.spec.ts (hunk 2 above)
Comment (DieselTripleX) [reply packages/ui/src/services/color-config/tests/color-functions.spec.ts]: Got it!

**Comment by m0ksem** [comment]
This function should be used somewhere in the framework. It would be nice, if you add this to useColors hook.

Then a user can use it in his components.

```
packages/ui/src/services/color-config/color-config.ts:50```

**Review by m0ksem — APPROVED** [review]

**Pull request merged**